api/tests/test_rate_limit.py

```diff
@@ -406,3 +406,11 @@ async def test_release_connection_on_exception(rate_limiter: RateLimiter) -> Non
 
     # Should be allowed because the connection should be freed after the exception
     await rate_limiter.acquire_connection('user-last', 'room-1')
+
```

Comment:
I think we need tests for zero connections and tests for connections spread across multiple users